### PR TITLE
Master adrv9009 dac dma example

### PR DESCRIPTION
**Pull request by stefpopa**
This patch offers an example of how to send custom data using the DMA.
The custom data is a sine wave stored in the sine_lut_iq array. Each
element of the array is 32-bit word composed of 16 bits of I data and 16
bits of Q data. I and Q parts are 90 degrees of of phase.

### CODE REVIEW

File: projects/adrv9009/src/app/headless.c [hunk 1]
```diff
@@ -760,13 +772,30 @@ int main(void)
 	/* Initialize the ADC core */
 	axi_adc_init(&rx_adc, &rx_adc_init);
 
+#ifdef DAC_DMA_EXAMPLE
+	gpio_get(&gpio_plddrbypass, DAC_FIFO_BYPASS);
+	gpio_direction_output(gpio_plddrbypass, 1);
+	axi_dac_load_custom_data(tx_dac,
+			sine_lut_iq,
+			sizeof(sine_lut_iq) / sizeof(uint32_t),
+			XPAR_DDR_MEM_BASEADDR + 0xA000000);
```
Comment (dbogdan) [inline projects/adrv9009/src/app/headless.c]: XPAR_DDR_MEM_BASEADDR is Xilinx dependent. DDR_MEM_BASEADDR should be used instead - it is defined in projects/adrv9009/src/devices/adi_hal/parameters.h .

File: projects/drivers/axi_dac_core/axi_dac_core.c [hunk 2]
```diff
@@ -599,6 +810,44 @@ int32_t axi_dac_set_buff(struct axi_dac *dac,
...
+					custom_data_iq[index]);
+#else
+			Xil_Out32(address + index_mem * sizeof(uint32_t),
+					custom_data_iq[index]);
+#endif
+			index_mem++;
+		}
+	}
+	Xil_DCacheFlush();
+
```
Comment (dbogdan) [inline projects/drivers/axi_dac_core/axi_dac_core.c]: Xil_DCacheFlush() is Xilinx dependent.

**Comment by stefpopa** [comment]
Changes in v2:
* Fixed astyle warnings
* Replaced `XPAR_AXI_ADRV9009_TX_DMA_BASEADDR `with `TX_DMA_BASEADDR`
* Replaced  `XPAR_DDR_MEM_BASEADDR `with `DDR_MEM_BASEADDR`
* Added a compiler switch around `Xil_DCacheFlush()`

**Comment by stefpopa** [comment]
Changes in v3:
* New commit which adds the `DMA_LAST` flag

**Pull request merged**